core/modules/css/KvInterfaceImplMySql.cc

```diff
@@ -0,0 +1,503 @@
+// -*- LSST-C++ -*-
+
+/*
+ * LSST Data Management System
+ * Copyright 2015 AURA/LSST.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+
+/**
+  * @file KvInterfaceImplMySql.h
+  *
+  * @brief Interface to the Central State System - MySql-based implementation.
+  *
+  * @Author Nathan Pease, SLAC
+  */
+
+
+// Class header
+#include "css/KvInterfaceImplMySql.h"
+
+// System headers
+#include <fstream>
+#include <iostream>
+#include <sstream>
+#include <stdexcept>
+
+// Third-party headers
+#include "boost/algorithm/string.hpp"
+#include "boost/algorithm/string/predicate.hpp"
+#include "boost/algorithm/string/split.hpp"
+#include "boost/format.hpp"
+#include "boost/lexical_cast.hpp"
+#include "mysql/mysqld_error.h"
+
+// LSST headers
+#include "lsst/log/Log.h"
+
+// Qserv headers
+#include "css/CssError.h"
+#include "sql/SqlResults.h"
+#include "sql/SqlTransaction.h"
+
+using std::map;
+using std::string;
+using std::vector;
+
+namespace lsst {
+namespace qserv {
+namespace css {
+
+namespace {
+
+    // logger instance for this module
+    LOG_LOGGER _logger = LOG_GET("lsst.qserv.css.KvInterfaceImplMySql");
+    const char KEY_PATH_DELIMITER('/');
+
+    /**
+     * @brief Helper for getting the value from an INT row
+     * @param results the object containing results information
+     * @param value the output var to return the result
+     * @return true if an int result could be obtained else false
+     */
+    bool extractIntValueFromSqlResults(sql::SqlResults* results, unsigned int* value) {
```

Comment:
SqlResults::extractFirstValue is not const so the short answer is no.
I could look into making the extract... methods in SqlResults const (or the ones that can be), but I had decided not to get too much into that. Should I do it?